fake-snippets-api/routes/api/snippets/list.ts

```diff
@@ -1,25 +1,53 @@
 import { withRouteSpec } from "fake-snippets-api/lib/middleware/with-winter-spec"
 import { z } from "zod"
 import { snippetSchema } from "fake-snippets-api/lib/db/schema"
+import type { DbClient } from "fake-snippets-api/lib/db/db-client"
+
+const getAccountByGithubUsername = (db: DbClient, username: string) => {
+  return db.accounts.find((a) => a.github_username === username)
+}
+
+const checkIsStarred = (
+  db: DbClient,
+  githubUsername: string | undefined,
+  packageId: string,
+) => {
+  if (!githubUsername) return false
+  const account = getAccountByGithubUsername(db, githubUsername)
+  return account ? db.hasStarred(account.account_id, packageId) : false
+}
 
 export default withRouteSpec({
   methods: ["GET", "POST"],
-  auth: "none",
+  auth: "optional_session",
   commonParams: z.object({
     owner_name: z.string().optional(),
     unscoped_name: z.string().optional(),
+    starred_by: z.string().optional(),
   }),
   jsonBody: z.any().optional(),
   jsonResponse: z.object({
     ok: z.boolean(),
     snippets: z.array(snippetSchema),
   }),
 })(async (req, ctx) => {
-  const { owner_name, unscoped_name } = req.commonParams
+  const { owner_name, unscoped_name, starred_by } = req.commonParams
+
+  const packages = ctx.db.packages.filter((pkg) => {
+    if (!pkg.is_snippet) return false
 
-  const packages = ctx.db
-    .getPackagesByAuthor(owner_name)
-    .filter((s) => !unscoped_name || s.unscoped_name === unscoped_name)
+    if (owner_name && pkg.owner_github_username !== owner_name) return false
+
+    if (unscoped_name && pkg.unscoped_name !== unscoped_name) return false
+
+    if (starred_by) {
+      const account = getAccountByGithubUsername(ctx.db, starred_by)
+      if (!account) return false
+      if (!ctx.db.hasStarred(account.account_id, pkg.package_id)) return false
+    }
+
+    return true
```

Comment:
@Anshgrover23 this is a bit nicer than your approach where you mutated a packages object imo, also demonstrates @kom-senapati didn't copy